src/shm/pool/multi.rs

```diff
@@ -251,26 +301,66 @@ impl<K: PartialEq + Clone> MultiPool<K> {
                     Some(Arc::new(BufferObjectData { free: free.clone() })),
                 )
                 .ok()?;
-            buffer = Proxy::from_id(conn, buffer_id).ok()?;
-            self.buffer_list.push(BufferHandle {
-                offset,
-                used: 0,
-                free,
-                buffer: Some(buffer.clone()),
-                size,
-                key: key.clone(),
-            });
-        } else {
-            return None;
+            buf_slot.free = free;
+            buf_slot.buffer = Proxy::from_id(conn, buffer_id).ok();
         }
-
-        let slice = &mut self.inner.mmap()[offset..][..size];
-
-        self.buffer_list[index?].free.swap(false, Ordering::Relaxed);
-
-        Some((offset, buffer, slice))
+        buf_slot.free.store(false, Ordering::Relaxed);
+        let buf = buf_slot.buffer.as_ref().unwrap();
+        Some((offset, buf, &mut self.inner.mmap()[offset..][..size]))
+    }
+    /// Calcule the offet and size of a buffer based on its stride.
+    fn offset(&self, mut offset: i32, width: i32, stride: i32, height: i32) -> (usize, usize) {
+        // bytes per pixel
+        let bpp = (stride as f32 / width as f32).ceil() as i32;
+        let size = stride * height;
+        // 5% padding.
+        offset += offset / 20;
+        offset += offset % bpp;
+        offset -= offset % bpp;
+        (offset as usize, size as usize)
+    }
+    /// Resizes the pool and appends a new buffer.
+    fn dyn_resize(
```

Comment:
Clippy complains, given this is a privte method, lets add a `#[allow(clippy::too-many-arguments)]` here I'd say.